tests/integration/targets/custom_image/tasks/tests.yml

```
@@ -0,0 +1,267 @@
+  assert:
+    that:
+      - failed_import.msg.startswith('Cannot import a new image without url.')
+
+- name: Import a custom image and wait for import
+  cloudscale_ch.cloud.custom_image:
+    name: "{{ cloudscale_resource_prefix }}-test-image"
+    state: present
+    slug: custom-ansible-image
+    url: "{{ image_url }}"
+    zones: lpg1
+    user_data_handling: 'pass-through'
+    tags:
+      project: mars
+    source_format: raw
+  register: image1
+  retries: 15
+  delay: 5
+  until: image1.import_status == 'success'
+  failed_when: image1.import_status == 'failed'
+- name: Verify import a custom image and wait for import
+  assert:
+    that:
+      - image1.import_status == 'success'
+      - image1.name == "{{ cloudscale_resource_prefix }}-test-image"
+
+- name: Import a custom image (idempotency)
+  cloudscale_ch.cloud.custom_image:
+    name: "{{ cloudscale_resource_prefix }}-test-image"
+    state: present
+    slug: custom-ansible-image
+    url: "{{ image_url }}"
+    zones: lpg1
+    user_data_handling: 'pass-through'
+    tags:
+      project: mars
+    source_format: raw
+  register: image
+- name: Verify import a custom image (idempotency)
+  assert:
+    that:
+      - image is not changed
+      - image.name == "{{ cloudscale_resource_prefix }}-test-image"
+      - image.uuid == image1.uuid
+
+- name: Import a custom image with bad url
+  cloudscale_ch.cloud.custom_image:
+    name: "{{ cloudscale_resource_prefix }}-test-image2"
+    state: present
+    slug: custom-ansible-image
+    url: "{{ image_url }}-non-existent"
+    zones: lpg1
+    user_data_handling: 'pass-through'
+    tags:
+      project: mars
+    source_format: raw
+  register: failed_import
+  ignore_errors: true
+  until: failed_import.import_status == 'failed'
+- name: Verify Fail import a custom image with bad url
+  assert:
+    that:
+      - failed_import.error_message.startswith('Expected HTTP 200, got HTTP 40')
+
+- name: Import a custom image and a failed import with that name exists
+  cloudscale_ch.cloud.custom_image:
+    name: "{{ cloudscale_resource_prefix }}-test-image2"
+    state: present
+    slug: custom-ansible-image
+    url: "{{ image_url }}"
+    zones: lpg1
+    user_data_handling: 'pass-through'
+    tags:
+      project: mars
+    source_format: raw
+  register: image2
+- name: Verify import a custom image
+  assert:
+    that:
+      - image2 is changed
+      - image2.name == "{{ cloudscale_resource_prefix }}-test-image2"
+      - image2.uuid != image1.uuid
+
+- name: Wait for import
+  cloudscale_ch.cloud.custom_image:
+    uuid: "{{ image2.uuid }}"
+  retries: 15
+  delay: 5
+  register: import_status
+  until: import_status.import_status == 'success'
+  failed_when: import_status.import_status == 'failed'
+- name: Verify Wait for import
+  assert:
+    that:
+      - import_status is not changed
+      - import_status.name == "{{ cloudscale_resource_prefix }}-test-image2"
+
+- name: Get newest image by name
```

As managing more than one image with the same name is no longer supported:
```suggestion
- name: Get image by name
```